contracts/SponsorFaucet.sol

```diff
@@ -23,96 +23,176 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
     uint256 public collateral_bound;
     //upper bound for single tx
     uint256 public upper_bound;
-    
-    mapping(address=>detail) public dapps;
-    
-    event applied(address indexed applicant, address indexed dapp, uint256 indexed amount);
-    
-    SponsorWhitelistControl internal_sponsor = SponsorWhitelistControl(0x0888000000000000000000000000000000000001);
-    
-    constructor(uint256 gasTotalLimit, uint256 collateralTotalLimit, uint256 gasBound, uint256 collateralBound, uint256 upperBound) public {
-        require(upperBound.mul(1000) <= gasBound, 'upperBound too high');
+
+    mapping(address => detail) public dapps;
+
+    event applied(
+        address indexed applicant,
+        address indexed dapp,
+        uint256 indexed amount
+    );
+
+    SponsorWhitelistControl internal_sponsor = SponsorWhitelistControl(
+        0x0888000000000000000000000000000000000001
+    );
+
+    constructor(
+        uint256 gasTotalLimit,
+        uint256 collateralTotalLimit,
+        uint256 gasBound,
+        uint256 collateralBound,
+        uint256 upperBound
+    ) public {
+        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
         gas_total_limit = gasTotalLimit;
         collateral_total_limit = collateralTotalLimit;
         gas_bound = gasBound;
         collateral_bound = collateralBound;
         upper_bound = upperBound;
     }
-     
-    /*** Dapp dev calls ***/ 
+
+    /*** Dapp dev calls ***/
     function applyBoth(address dapp) public nonReentrant whenNotPaused {
-        if(_isAppliableForGas(dapp)) _applyForGas(dapp);
-        if(_isAppliableForCollateral(dapp)) _applyForCollateral(dapp);
+        if (_isAppliableForGas(dapp)) _applyForGas(dapp);
+        if (_isAppliableForCollateral(dapp)) _applyForCollateral(dapp);
     }
 
     function applyGas(address dapp) public nonReentrant whenNotPaused {
-        if(_isAppliableForGas(dapp)) _applyForGas(dapp);
+        _validateApplyForGas(dapp);
+        _applyForGas(dapp);
     }
 
     function applyCol(address dapp) public nonReentrant whenNotPaused {
-        if(_isAppliableForCollateral(dapp)) 
+        _validateApplyForCollateral(dapp);
         _applyForCollateral(dapp);
     }
 
     function isAppliable(address dapp) public returns (bool) {
-        if(_isAppliableForGas(dapp) || _isAppliableForCollateral(dapp)) return true;
+        if (_isAppliableForGas(dapp) || _isAppliableForCollateral(dapp))
+            return true;
```

Comment:
why not merge into one line `return xxx || yyy;`

Reply:
In that way, **require** statements in validate functions will not be executed and no error message.